src/power_grid_model/core/serialization.py

```diff
@@ -94,10 +100,16 @@ class Serializer(ABC):
     _dataset: CConstDataset
     _serializer: SerializerPtr
 
-    def __new__(cls, data: Dataset, serialization_type: SerializationType, dataset_type: Optional[DatasetType] = None):
+    def __new__(cls, data: Dataset, serialization_type: SerializationType, dataset_type: DatasetType | None = None):
         instance = super().__new__(cls)
 
-        instance._data = data
+        if not data:
+            instance._data = {}
+        else:
+            dataset_type = dataset_type if dataset_type is not None else get_dataset_type(data)
+            instance._data = copy_to_row_or_columnar_dataset(
```

Comment:
why is a copy always required? it should only be required when dealing with columnar datasets. and even then, i don't think the copy should be done here but in the dataset itself, because that is something that will change when the API starts supporting columnar data

Reply:
If the input of copy_to_row_or_columnar_dataset is row based and requested output is row based too, a copy is not created. Ill add that to the docstring too.

Reply:
then the name is not good. maybe you can go with `compatibility_convert_to_row_based`

Reply:
It should not matter where this copy is being done since it is temporary. This was a place where changes were minimal now and later when we have to remove it.